src/openrct2/actions/StaffSetPatrolAreaAction.cpp

```diff
@@ -78,20 +70,52 @@ GameActions::Result::Ptr StaffSetPatrolAreaAction::Execute() const
 
     if (isPatrolling)
     {
-        gStaffModes[staff->StaffId] = StaffMode::Patrol;
+        gStaffModes[staff.StaffId] = StaffMode::Patrol;
     }
-    else if (gStaffModes[staff->StaffId] == StaffMode::Patrol)
+    else if (gStaffModes[staff.StaffId] == StaffMode::Patrol)
     {
-        gStaffModes[staff->StaffId] = StaffMode::Walk;
+        gStaffModes[staff.StaffId] = StaffMode::Walk;
     }
+}
 
+static void InvalidatePatrolTile(const CoordsXY& loc)
+{
     for (int32_t y = 0; y < 4 * COORDS_XY_STEP; y += COORDS_XY_STEP)
     {
         for (int32_t x = 0; x < 4 * COORDS_XY_STEP; x += COORDS_XY_STEP)
         {
-            map_invalidate_tile_full({ (_loc.x & 0x1F80) + x, (_loc.y & 0x1F80) + y });
+            map_invalidate_tile_full({ (loc.x & 0x1F80) + x, (loc.y & 0x1F80) + y });
```

Comment:
```suggestion
            map_invalidate_tile_full(loc + CoordsXY{ x, y });
```
The location needs to be aligned only once, best to make a copy before the loop.